docs/searching.rst

```diff
@@ -0,0 +1,348 @@
+
+1. Create a *query object* with the ``select()`` method.
+2. Refine the query by using the query object's methods to add a text *query* and/or *search criteria.*
+3. Execute the query to see its results.
+
+Creating a Query Object
+-----------------------
+
+A query object is created via the :func:`CBCloudAPI.select() <cbc_sdk.connection.BaseAPI.select>` operation, specifying
+the type of data to be retrieved.
+
+In this example, we create a query to search for all devices with antivirus active::
+
+    # assume the CBCloudAPI object is in the variable "api"
+    >>> from cbc_sdk.platform import Device
+    >>> device_query = api.select(Device).where('avStatus:AV_ACTIVE')
+
+    # The device query has been created but not yet executed
+    >>> type(device_query)
+    <class 'cbc_sdk.platform.devices.DeviceSearchQuery'>
+
+The ``select()`` method may take either a class reference or a string class name::
+
+    >>> query1 = api.select(Device)
+    >>> query2 = api.select("Device")
+
+    # prove that the query we get back in either case is the same
+    >>> type(query1) == type(query2)
+    True
+
+Selecting an Object Directly
+****************************
+
+The ``select()`` method may also be used to retrieve an object directly if you know its ID value, by passing the ID as
+a second parameter::
+
+    >>> dev = api.select(Device, 1234567)  # assume this device exists
+    >>> type(dev)
+    <class 'cbc_sdk.platform.devices.Device'>
+
+Refining a Query
+----------------
+
+Queries may support one of two different methods for refining a query:
+
+* Through the use of *text query.*
+* Through adding *criteria.*
+
+Text Query Support
+******************
+
+Text queries may be added to a query object by using the query object's
+:func:`where() <cbc_sdk.base.QueryBuilder.where>`, :func:`and_() <cbc_sdk.base.QueryBuilder.and_>`, and
+:func:`or_() <cbc_sdk.base.QueryBuilder.or_>` methods.  The following example sets up a query looking for events
+in which the program ``googleupdate.exe`` accesses the system registry on a device with a specific hostname, IP
+address, and owner::
+
+    # assume the CBCloudAPI object is in the variable "api"
+    from cbc_sdk.endpoint_standard import Event
+    >>> event_query = api.select(Event).where(hostName='Win10').and_(ipAddress='10.0.0.1')
+
+    # further refine the query
+    >>> event_query.and_(applicationName='googleupdate.exe').and_(eventType='REGISTRY_ACCESS')
+    >>> event_query.and_(ownerNameExact='DevRel')
+
+The ``where()`` method supplies the initial query parameters, while ``and_()`` and ``or_()`` add additional query
+parameters. As with other languages, ``and_()`` gets grouped together before ``or_()``.
+
+Parameters may either be supplied as text strings or as keyword assignments::
+
+    >>> from cbc_sdk.platform import Device
+    # the following two queries are equivalent
+    >>> string_query = api.select(Device).where("avStatus:AV_ACTIVE")
```

Comment:
avStatus is not a common used property can we use status instead?

Reply:
Again, I didn't write these examples, I just copied them out of "concepts." That said, I'm replacing each one of the "avStatus" values I see in this file with "status."

Reply:
You will also need to modify the value being used as the status variable has different values https://developer.carbonblack.com/reference/carbon-black-cloud/platform/latest/devices-api/#base-device

Reply:
I changed the "AV_ACTIVE" value to "ACTIVE." That should suffice.